Accomplish basic layer compatibility with oe/dunfell

amstewart opened the pull request:
This patchset establishes basic layer compatibility with the dunfell release of OE-core, specifically those rebased layer branches available at:
github.com/amstewart/$layer:dev/dunfell/master.

The recipe commits in this patchset allow a selection of essential tool recipes (eg. glibc, busybox, util-linux, et c.) to succeed. Further layer fixes are required for full dunfell compatibility, but that work will be tracked in other work-items and manifest in later pull requests.

@ni/rtos

amstewart committed:
layer.conf: assert layer compatibility with oe/dunfell

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
libarchive: remove recipe .patches

All .patches to the libarchive upstream have been accepted, as of tag
`v3.3.3`, and consumed by oe-core - which is now at v3.4.2 (OE/dunfell).
These recipe .bbappends should be dropped.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
busybox: reconcile appends with bbox 1.31

OE/Dunfell has upgraded busybox to 1.31.

* Bump the .bbappend version to apply atop oe-core.
* Reconcile the NILRT defconfig with the new version.
* Remove the 0001-zcip-fix-wrong....patch because it is available
  through upstream busybox.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
util-linux: remove cyclic DEPENDS on niacctbase

util-linux has a target-only DEPENDS on the niacctbase recipe (also in
meta-nilrt). This dependency is not actually required at build time and
creates a cyclic dependency that runs through shadow, libsemanage,
audit, and other recipes.

Remove the spurious dependency on niacctbase to break the loop.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
util-linux: fixup CLASSOVERRIDE DEPENDS

Specifying DEPENDS_class-target_appends (or variations otherof) results
in the internal value of DEPENDS from util-linux.bb being overwritten
entirely with the DEPENDS_class-target value. As a result, `libcap-ng`
is never added to the recipe sysroot and the util-linux:do_configure
task fails with an error that it cannot find libcap-ng.

Instead, use the syntax: `DEPENDS_append_class-target` to have bitbake
append to the `DEPENDS` variable in a non-native recipe build. This is
the accepted syntax in OE upstream.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
busybox: fixup SRC_URI machine override

Using the _append operation without including a space in the
value-string may cause the word to collide with previous words in the
SRC_URI string. Change the assignment operator to `+=` to have bitbake
automatically insert a space as the first element in the string-value.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
busybox: disable the MOUNT_NFS feature

The busybox feature CONFIG_FEATURE_MOUNT_NFS is only necessary in linux
kernels perior to 2.6. Further, this feature requires the rpc.h headers
from glibc<2.26, which have since been deprecated. As a result, busybox
upstream's recommendation is to disable this feature on newer kernels.

http://lists.busybox.net/pipermail/busybox/2017-September/085848.html

OE upstream has done the same in their busybox defconfig.

https://www.openembedded.org/pipermail/openembedded-core/2015-November/113155.html

Disable it in our defconfig for the same reasons.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
pkggrp-ni-base: replace start-stop-daemon

The start-stop-daemon recipe has been deprecated in upstream meta-oe in
favor of the `dpkg-start-stop` package from the `dpkg` recipe. It
allegedly provides a newer version of the same binary.

Replace packagegroup-ni-base references to `start-stop-daemon` with the
new dpkg package RDEPENDS.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
nilrt.conf: rev distro version to 9.0

Update the DISTRO_VERSION to 9.0; service for the LV 2021 release
schedule.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

shruthi-ravi reviewed (commented):
Are the cp936 patches here related to the condvar patches?

amstewart commented:
> Are the cp936 patches here related to the condvar patches?

Hmm... I'm not sure, but I don't see anything which obviously says that they are.
@gratian Can you comment on if we should carry and reapply those few cp936 patches to the new glibc?

shruthi-ravi commented on recipes-core/busybox/busybox/defconfig:
```diff
@@ -78,10 +81,12 @@ CONFIG_NO_DEBUG_LIB=y
 # CONFIG_FEATURE_USE_BSS_TAIL is not set
+CONFIG_FLOAT_DURATION=y
 CONFIG_FEATURE_RTMINMAX=y
+CONFIG_FEATURE_RTMINMAX_USE_LIBC_DEFINITIONS=y
```
```suggestion
# CONFIG_FEATURE_RTMINMAX_USE_LIBC_DEFINITIONS is not set
```
based on this [file](https://github.com/openembedded/openembedded-core/blob/dunfell/meta/recipes-core/busybox/busybox/defconfig#L88). Or is this something you intended to do?

amstewart replied:
It was enabled by the `make oldconfig` operation in busybox because we had the parent `CONFIG_FEATURE_RTMINMAX` enabled already. Codesearching around for which signal definitions we use suggests that both NI and OE-upstream prefer to use the LIBC definitions. So we should keep this option enabled (which is the default anyway).

@gratian ACKed that assessment.

gratian commented:
> > Are the cp936 patches here related to the condvar patches?
> 
> Hmm... I'm not sure, but I don't see anything which obviously says that they are.
> @gratian Can you comment on if we should carry and reapply those few cp936 patches to the new glibc?

The cp936 patches are not related to condvar support. They are likely related to simplified Chinese localization support. We do support simplified Chinese localization on LV RT and as such they probably need to be carried forward.

shruthi-ravi commented on recipes-core/util-linux/util-linux_2.%.bbappend:
```diff
@@ -1,6 +1,6 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
-DEPENDS_class-target += "shadow-native pseudo-native niacctbase"
```
Would you get an error if you tried to build the recipe with a missing dependency?

amstewart replied:
Presumably, a build-time dependency which you actually need would cause some kind of error at build. The commits which added the original DEPENDS didn't go into detail about why `niacctbase` was necessary. I would assume it was thought to be, since the `niacctbase` recipe does some configuration of the lvrt user and NI-specific groups.

But I don't believe that recipe is a real dependency - the do_install operations in util-linux seem independent of what `niacctbase` is doing in the sysroot.

shruthi-ravi commented on recipes-core/busybox/busybox/defconfig:
```diff
@@ -658,7 +658,7 @@ CONFIG_MOUNT=y
 # CONFIG_FEATURE_MOUNT_HELPERS is not set
 # CONFIG_FEATURE_MOUNT_LABEL is not set
-CONFIG_FEATURE_MOUNT_NFS=y
+CONFIG_FEATURE_MOUNT_NFS=n
```
```suggestion
# CONFIG_FEATURE_MOUNT_NFS is not set
```

amstewart replied:
The default value for this feature (which doesn't apply for kernels >2.6) is 'no'. So your suggestion seems like a NOOP. Why do you think it should be "not set" rather than explicitly 'no'?

shruthi-ravi commented on recipes-core/packagegroups/packagegroup-ni-base.bb:
```diff
@@ -118,7 +119,6 @@ RDEPENDS_${PN} = "\
 	opkg-keyrings \
 	os-release \
 	run-postinsts \
-	start-stop-daemon \
```
codesearch reveals that a lot of people use start-stop-daemon! Are you making a list of breaking changes that we can tell clients upfront? This should be added to that.

ScotSalmon replied:
Um, yeah. We're really not supporting that anymore? That will break a ton of stuff.

amstewart replied:
The `start-stop-daemon` binary is still provided, just through the `dpkg-start-stop` package in oe-core, rather than the `start-stop-daemon` recipe in meta-oe. Meta-oe removed their recipe because the one in oe-core is newer (and from the same source).

So initscripts can continue to use `start-stop-daemon` in the same way. The only impact is if they have declared an IPK dependency on `start-stop-daemon`, it will need to be updated to `dpkg-start-stop`.

shruthi-ravi commented on conf/distro/nilrt.conf:
```diff
@@ -2,9 +2,9 @@ require nilrt.inc
 
```
I suppose you won't be able to edit the commit message to say LV 2021 now?

amstewart replied:
Oops. I'll fix that up in my v2. :)

amstewart committed:
glibc: remove 2.24 patches

OE/dunfell has upgraded to glibc 2.31 and new librtpi library obviates
the need for the condvar patchset. Remove these patches and accept
upstream's glibc recipe.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
glibc: refresh cp936 patches

Refresh the NI-specific simplified Chinese (CP936) patches atop
glibc_2.31 from upstream oe/dunfell.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
# Patch V2
* Updated `glibc: remove 2.24 patches` to only remove the condvars patchset.
* Added a new glibc commit to refresh the cp936 patches against glibc_2.31.
* Fixed typo in `nilrt.conf: rev distro version to 9.0`.

amstewart committed:
busybox: unset apparently unused defconfig values

There are many values in the NILRT-busybox defconfig which have been
inherited from antiquated releases, frequently without proper historic
justification.

For some of these values which don't seem to serve an obvious purpose,
prefer to unset them and inherit our configuration from upstream.

CONFIG_TELINIT_PATH
* NILRT doesn't use telinit from busybox anyway, and the blank value
  being set here isn't sane.

CONFIG_FEATURE_KILL_DELAY
* NILRT doesn't use FEATURE_KILL_REMOVED, so there isn't an obvious
  reason why we assert this dependent config value.

CONFIG_INIT_TERMINAL_TYPE
* There doesn't seem to be a historic reason why we have used a blank
  value here, instead of the 'linux' default. Prefer the upstream
  defaults.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
# Patch V3
* Added `busybox: unset apparently unused defconfig values` to address feedback from @shruthi-ravi about unsetting unused values.

shruthi-ravi reviewed (approved):
Looks good!
